examples/ginkgo/ex1.cpp

```diff
@@ -86,9 +88,16 @@ int main(int argc, char *argv[])
    args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
                   "--no-visualization",
                   "Enable or disable GLVis visualization.");
-   args.AddOption(&use_ginkgo_solver, "-gko", "--use_gko_solver", "-no-gko",
+   args.AddOption(&use_ginkgo_solver, "-gko-solver", "--use-gko-solver",
+                  "-no-gko-solver",
```

Comment:
What do you think about replacing both of these with an integer parameter (`-s`?) which will correspond to different combinations, like 0 = Ginkgo solver and preconditioner, 1 = Ginkgo solver and MFEM preconditioner, 2 = MFEM solver and Ginkgo preconditioner, 3 = MFEM solver and preconditioner?  Maybe this would help get rid of some of the nested ifs you mentioned, as well?

Reply:
That's also fine by me — up to you.

Reply:
I tried something like this in 2bd0031b8123582765d893b84fc04aaf58dd4260.  I also made it output the solver/preconditioner combo for clarity, and give an error if trying to choose an option with a Ginkgo preconditioner in partial assembly mode.

Reply:
Looks good to me!